src/Service/DocumentUploader.php

```diff
@@ -38,6 +39,41 @@ public function __construct(SluggerInterface $slugger, string $uploadDocumentDir
         $this->mimeTypeGuesser = $mimeTypeGuesser;
     }
 
+    /**
+     * Creates and returns new document from filename.
+     */
+    public function createDocumentFromPath(string $fileName, User $user, string $documentName, string $documentType): Document
+    {
+        $document = new Document();
+
+        $document->setFilename($fileName);
+        $document->setDocumentName($documentName);
+        $document->setPath($this->getFilepathFromProjectDirectory($fileName));
+        $document->setUploadedBy($user);
+        $document->setType($documentType);
+
+        return $document;
+    }
+
+    /**
+     * Creates, uploads and returns new document from a file.
+     */
+    public function createDocumentFromFile(UploadedFile $file, User $user, string $documentName, string $documentType): Document
+    {
+        $document = new Document();
+
+        $document->setDocumentName($documentName);
+        $document->setType($documentType);
+        $document->setOriginalFileName($file->getClientOriginalName());
+        $document->setUploadedBy($user);
+
+        $newFileName = $this->upload($file);
```

Comment:
You could set this at the top of the function body and then return the result of calling `createDocumentFromPath`:

```php
$path = $this->upload($file);
$document = $this->createDocumentFromPath($path, …);
$document->setOriginalFileName($file->getClientOriginalName());

return $document;
```

Reply:
👍